README.md

```diff
@@ -0,0 +1,94 @@
+
+E2E (in `e2e`) contains a test that uses the client to talk to the server to read write and read to/from the test DynamoDB instance. That test also requires the 'dev/test environment' to be up.
+
+
+## Using
+
+### Server
+The easiest way to use the server is to use the docker image from the [Snowplow docker repo][snowplow-docker].
+
+Alternatively see below for an [example server run](#example-run-server).
+
+### Client
+You can either use the client library from this project if it suits your needs, or generate your own client code using GRPC and use it. For an example of using the code form this project, see [snowplow-piinguin-relay][piinguin-relay].
+
+## Get dev/test environment up:
+
+```
+vagrant up
+...
+vagrant ssh
+```
+
+## Run end-to-end test:
+```
+sbt e2e/test
+```
+
+## Example run server:
+
+```
+$ sbt "server/run -h 0.0.0.0 -p 8080 -t piinguin"
+
+```
+
+## Example client usage (the table must exist on dynamo to successfully write):
+
+```
+
+$ sbt "client/console"
+
+scala> import scala.concurrent.{ExecutionContext, Await}
+import scala.concurrent.{ExecutionContext, Await}
+
+scala> import scala.concurrent.duration._
+import scala.concurrent.duration._
+
+scala> import com.snowplowanalytics.piinguin.client.PiinguinClient
+import com.snowplowanalytics.piinguin.client.PiinguinClient
+
+scala> implicit val ec = ExecutionContext.global
+ec: scala.concurrent.ExecutionContextExecutor = scala.concurrent.impl.ExecutionContextImpl@7f4b9eca
+
+scala> val c = new PiinguinClient("localhost", 8080)
+c: com.snowplowanalytics.piinguin.client.PiinguinClient = com.snowplowanalytics.piinguin.client.PiinguinClient@6e5990ce
+
+scala> val createResult = Await.result(c.createPiiRecord("123", "456"), 10 seconds)
+
+createResult: Either[com.snowplowanalytics.piinguin.client.FailureMessage,com.snowplowanalytics.piinguin.client.SuccessMessage] = Left(FailureMessage(UNAVAILABLE: io exception))
+
+or 
+
+createResult: Either[com.snowplowanalytics.piinguin.client.FailureMessage,com.snowplowanalytics.piinguin.client.SuccessMessage] = Left(FailureMessage(Cannot do operations on a non-existent table (Service: AmazonDynamoDBv2; Status Code: 400; Error Code: ResourceNotFoundException; Request ID: 287894d3-ced6-4382-ab9b-bb4f51a0070c)))
+
+or 
+
+createResult: Either[com.snowplowanalytics.piinguin.client.FailureMessage,com.snowplowanalytics.piinguin.client.SuccessMessage] = Right(SuccessMessage(OK))
+
+scala> import com.snowplowanalytics.piinguin.server.generated.protocols.piinguin.ReadPiiRecordRequest.LawfulBasisForProcessing
+scala> val readResult = Await.result(c.readPiiRecord("123", LawfulBasisForProcessing.CONSENT), 10 seconds)
+
+                                                                   ^
+readResult: Either[com.snowplowanalytics.piinguin.client.FailureMessage,com.snowplowanalytics.piinguin.client.PiinguinClient.PiiRecord] = Right(PiiRecord(123,456))
+
+```
+
```

Comment:
you need a copyright and license section at the end, e.g. https://github.com/snowplow/scala-maxmind-iplookups